pkg/synchromanager/clustersynchro/resource_synchro.go

```diff
@@ -154,20 +194,12 @@ func (synchro *ResourceSynchro) Run(stopCh <-chan struct{}) {
 
 	status = clusterv1alpha2.ClusterResourceSyncCondition{
 		Status:             clusterv1alpha2.SyncStatusStop,
+		Reason:             "Pasue",
 		LastTransitionTime: metav1.Now().Rfc3339Copy(),
 	}
 	synchro.status.Store(status)
```

Comment:
The Reason for Pause is assumed to be executed before the synchro.status.Store operation in Run method? In fact the order is not guaranteed. And the synchro.status.Store operation here is not guaranteed to happen before the updateStatus operation in ClusterSynchro Shutdown method, because there is no waitGroup waiting for it.

Reply:
```
Status:
  Conditions:
    Last Transition Time:  2022-04-21T13:08:10Z
    Message:
    Reason:                Running
    Status:                True
    Type:                  ClusterSynchroInitialized
    Last Transition Time:  2022-04-22T14:04:17Z
    Message:
    Reason:                ClusterSynchroStop
    Status:                Unknown
    Type:                  Ready
  Sync Resources:
    Group:  apps
    Resources:
      Kind:        Deployment
      Name:        deployments
      Namespaced:  true
      Sync Conditions:
        Last Transition Time:  2022-04-22T14:04:17Z
        Reason:                Pasue
        Status:                Stop
        Storage Version:       v1
        Version:               v1
    Group:
    Resources:
      Kind:        Pod
      Name:        pods
      Namespaced:  true
      Sync Conditions:
        Last Transition Time:  2022-04-22T14:04:17Z
        Status:                Stop
        Storage Version:       v1
        Version:               v1
  Version:                     v1.23.3

```